### PR TITLE
Add duck db support and integration tests

### PR DESCRIPTION
**Pull request by turbo1912**

### CODE REVIEW

File: tests/mock/mockProfile/profiles.yml [hunk 1]
```diff
@@ -29,3 +29,9 @@ fal_test:
       dbname: test
       schema: dbt_fal
       threads: 4
+
+duckdb_dbt_project:
+  outputs:
+    default:
+      type: duckdb
+      path: /Users/gorkemyurtseven/dev/duck_db_database/db.duckdb
```
Comment (mederka) [inline tests/mock/mockProfile/profiles.yml:37]: Do we need `db.duckdb` committed?

Also, need to fix the path.

File: tests/test_cli.py [hunk 2]
```diff
@@ -7,7 +7,8 @@
 import re
 
 profiles_dir = os.path.join(Path.cwd(), "tests/mock/mockProfile")
-project_dir = os.path.join(Path.cwd(), "tests/mock")
+# project_dir = os.path.join(Path.cwd(), "tests/mock")
+project_dir = "/Users/gorkemyurtseven/dev/duck_db_project/duckdb_dbt_project"
```
Comment (mederka) [inline tests/test_cli.py:11]: 🤨

**Review by mederka — CHANGES_REQUESTED** [review]
So it works almost "out-of-the-box"? Awesome!

File: tests/test_cli.py (hunk 2 above)
Comment (turbo1912) [reply tests/test_cli.py:11]: i was just running the tests sorry not ready for review!

File: src/faldbt/lib.py [hunk 3]
```diff
@@ -375,6 +375,10 @@ def _alchemy_engine(adapter: SQLAdapter, database: Optional[str]):
         url_string = f"bigquery://{database}"
     if adapter.type() == "postgres":
         url_string = "postgresql://"
+
+    if adapter.type() == "duckdb":
+        url_string = "duckdb://"
+
```
Comment (chamini2) [inline src/faldbt/lib.py:381]: this is not necessary, by default we do 
```
url_string = f"{adapter.type()}://"
```

File: .github/workflows/test_profiles.yml [hunk 4]
```diff
@@ -33,6 +33,11 @@ jobs:
         working-directory: integration_tests
         run: pip install behave
 
+      - name: Install duckdb
+        working-directory: integration_tests
+        if: matrix.profile == 'duckdb'
+        run: pip install duckdb
```
Comment (chamini2) [inline .github/workflows/test_profiles.yml:39]: if fal installs `duckdb-engine`, do we need this?

**Review by chamini2 — CHANGES_REQUESTED** [review]

File: .github/workflows/test_profiles.yml [hunk 5]
```diff
@@ -65,6 +70,8 @@ jobs:
           SF_DATABASE: ${{ secrets.SF_DATABASE }}
           SF_WAREHOUSE: ${{ secrets.SF_WAREHOUSE }}
           SF_SCHEMA: ${{ secrets.SF_SCHEMA }}
+          # Duckdb
+          DB_PATH: ${{ github.workspace }}/temp/duck.db
```
Comment (chamini2) [inline .github/workflows/test_profiles.yml:74]: if we remove the `/temp` part we can remove the `mkdir temp` step right? Can we do that?

**Review by chamini2 — APPROVED** [review]
Great! Just one idea to consider

File: .github/workflows/test_profiles.yml (hunk 5 above)
Comment (turbo1912) [reply .github/workflows/test_profiles.yml]: yea simpler

**Pull request merged**